src/code_0807CC3C.c

```diff
@@ -0,0 +1,543 @@
+
+    u32 retval;
+    struct_0807D1C4* ptr;
+    u32 e0, e1;
+    u16 l1prep;
+
+    ptr = sub_0807D1C4(arg0);
+
+    thing.unk_3 = 'MCZ3';
+    l1prep = sub_0807D1A4((u16*)&thing.unk_3, 4);
+    l1prep += sub_0807D1A4((u16*)arg1, ptr->field_0x0);
+    thing.unk_1 = l1prep;
+    thing.unk_2 = -(u32)l1prep;
+    e0 = sub_0807D20C(ptr->field_0x6, (const char*)arg1, ptr->field_0x0);
+    if (e0) {
+        e0 = sub_0807D184(ptr->field_0x2, (const char*)&thing.unk_1);
+    }
+    e1 = sub_0807D20C(ptr->field_0x8, (const char*)arg1, ptr->field_0x0);
+    if (e1) {
+        e1 = sub_0807D184(ptr->field_0x4, (const char*)&thing.unk_1);
+    }
+
+    retval = 0;
+    if (e0 || e1) {
+        retval = 1;
+    }
+    return retval;
+}
+
+NONMATCH("asm/non_matching/code_0807CC3C/sub_0807D008.inc", u32 sub_0807D008(u32 param_1, SaveFile* saveFile)) {
+    u32 set_0;
+    char auStack32[8];
+
+    struct_0807D1C4* unk_s;
+    u32 t1;
+    u32 t2;
+    u32 ret;
+    u32 temp;
+
+    unk_s = sub_0807D1C4(param_1);
+    t1 = sub_0807D0EC(unk_s->field_0x2, auStack32);
+    if (t1 == 2) {
+        if ((sub_0807D1D8(unk_s->field_0x6, (char*)saveFile, unk_s->field_0x0) == 0) ||
+            (sub_0807D0A0((u16*)auStack32, (u16*)saveFile, (u32)unk_s->field_0x0) == 0)) {
+            t1 = 0;
+        } else {
+            return 1;
+        }
+    }
+    t2 = sub_0807D0EC(unk_s->field_0x4, auStack32);
+    if (t2 == 2) {
+        if ((sub_0807D1D8(unk_s->field_0x8, (char*)saveFile, unk_s->field_0x0) != 0) &&
+            (sub_0807D0A0((u16*)auStack32, (u16*)saveFile, (u32)unk_s->field_0x0) != 0)) {
+            return 1;
+        }
+        t2 = 0;
+    }
+    set_0 = 0;
+    CpuSet(&set_0, saveFile, unk_s->field_0x0 >> 2 | CPU_SET_SRC_FIXED | CPU_SET_32BIT);
+    temp = t1 | t2;
+    ret = 0;
+    if (temp == 0) {
+        ret = -1;
+    }
+    return ret;
+}
+END_NONMATCH
+
+NONMATCH("asm/non_matching/code_0807CC3C/sub_0807D0A0.inc", u32 sub_0807D0A0(u16* unk_1, u16* unk_2, u32 unk_3)) {
+    u32 r0;
+
+    u32* u32_1 = (u32*)unk_1;
+
+    u16 u0;
+    u32 u1;
+
+    u0 = sub_0807D1A4(unk_1 + 2, 4);
+    u0 = u0 + sub_0807D1A4(unk_2, unk_3);
+
+    u1 = unk_1[0];
+    if (u1 != u0) {
+        r0 = 0;
+    } else {
+        if (unk_1[1] == (-(u1 << 0x10) >> 0x10)) {
+            if (*(u32_1 + 1) != 'MCZ3') {
+                r0 = 0;
+            } else {
+                r0 = 1;
+            }
+        } else {
+            r0 = 0;
+        }
+    }
+    return r0;
+}
+END_NONMATCH
+
+u32 sub_0807D0EC(u32 unk_1, const char* unk_2) {
+    u32 ret;
+
+    if (!sub_0807D1D8(unk_1, unk_2, 8)) {
+        ret = 0;
+    } else {
+        ret = sub_0807D128((Thing*)unk_2);
+    }
+    if (!ret && sub_0807D1D8(unk_1 + 8, unk_2, 8)) {
+        ret = sub_0807D128((Thing*)unk_2);
+    }
+    return ret;
+}
+
+u32 sub_0807D128(const Thing* thing) {
+    u32 ret;
+    switch (thing->unk_3) {
+        case 'MCZ3':
+            if (thing->unk_1 + thing->unk_2 == 0x10000) {
```

Comment:
Are these u16s part of the checksum?

Maybe chksum_lo and chksum_hi

Reply:
they are, will also be done with that new header struct